### PR TITLE
Add base template for Protocol design system (Fixes #5752)

### LINKED ISSUE #5752: Create bedrock base template for Protocol
This is an issue to track creation of a new base template for using Protocol in bedrock. There are still a couple of global pieces missing from Protocol, such as the main nav/footer, and the download button. Until these pieces exist in Protocol, we may need to reuse some Pebbles component CSS as a temporary measure.

### PR DESCRIPTION
**Pull request by alexgibson**
## Description
- Adds a new base template for Protocol.
- Ports the following components which do not yet exist in Protocol:
  - Global nav
  - Legacy masthead
  - Download button
  - Footer
- Adds a universal Protocol stylesheet for legacy IE browsers.
- Adds temporary test page for demo purposes.

## Issue / Bugzilla link
https://github.com/mozilla/bedrock/issues/5752

## Testing
Demo page: https://bedrock-demo-agibson.oregon-b.moz.works/en-US/protocol-test/

### CODE REVIEW

File: bedrock/mozorg/urls.py [hunk 1]
```diff
@@ -315,4 +315,6 @@
     url(r'^microsummaries/0\.1$', views.namespaces, {'namespace': 'microsummaries'}),
     url(r'^projects/xforms/2005/type$', views.namespaces, {'namespace': 'xforms-type'}),
     url(r'^xbl$', views.namespaces, {'namespace': 'xbl'}),
+
+    page('protocol-test', 'mozorg/protocol.html'),
```
Comment (alexgibson) [inline bedrock/mozorg/urls.py]: TODO: remove this page post-code review.

File: media/css/protocol/components/_download-button.scss [hunk 2]
```diff
@@ -0,0 +1,234 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+@import "../../../../node_modules/@mozilla-protocol/core/protocol/css/includes/lib";
```
Comment (craigcook) [inline media/css/protocol/components/_download-button.scss]: Maybe I'm crazy, but after a little local experimenting I'm not sure we actually need to re-import this in every file. It's become such a habit for so long I haven't even questioned it before, but now I wonder if it's a legacy from Sandstone/Less that isn't needed in Sass. It needs more thorough testing. But I don't like the length and fragility of this relative path so if we don't have to repeat it everywhere that would be nice. Not relevant to this PR but something to think about.

Comment (alexgibson) [reply media/css/protocol/components/_download-button.scss]: Godd point - let's investigate further in Protocol itself 👍

**Comment by alexgibson** [comment]
Thanks @craigcook - demo page removed :shipit:

**Pull request merged**